esmvalcore/preprocessor/_regrid.py

```diff
@@ -469,13 +472,25 @@ def regrid(cube, target_grid, scheme, lat_offset=True, lon_offset=True):
 
     # Return non-regridded cube if horizontal grid is the same.
     if not _horizontal_grid_is_close(cube, target_grid):
+        original_dtype = cube.core_data().dtype
 
-        # Perform the horizontal regridding.
+        # Perform the horizontal regridding
         if _attempt_irregular_regridding(cube, scheme):
             cube = esmpy_regrid(cube, target_grid, scheme)
         else:
             cube = cube.regrid(target_grid, HORIZONTAL_SCHEMES[scheme])
 
+        # Preserve dtype for 'unstructured_nearest' scheme
+        if scheme == 'unstructured_nearest':
+            try:
+                cube.data = cube.core_data().astype(original_dtype,
+                                                    casting='same_kind')
+            except TypeError as exc:
+                logger.warning(
```

Comment:
```suggestion
                logger.debug(
```
Does this happen often? Is this really important info for the user?

Reply:
Quite right, see above: https://github.com/ESMValGroup/ESMValCore/pull/1079#discussion_r710075266.
But that's beyond us here. Let's keep the warning for now and try to fix upstream in the long term.